### PR TITLE
CA3075: Fix some false positive cases with XmlReader.Create() overloads

### LINKED ISSUE #1915: CA3075 appears to have false positives with XmlReader.Create(TextReader) and XmlReader.Create(Stream) overloads
But when I change @dotpaul 's example to use the proposed XmlReader.Create factory methode I get the same finding:

```CSharp
static void Main(string[] args)
{
    string hahaXml = @"<?xml version=""1.0""?>
        <!DOCTYPE lolz [
        <!ENTITY ha ""ha"">
        <!ENTITY ha2 ""&ha;&ha;"">
        ]>
        <lolz> &ha2; </lolz>";
    var x = XmlReader.Create(new StringReader(hahaXml));
    while (x.Read())
    {
        if (x.NodeType == XmlNodeType.Element)
        {
            Console.WriteLine(x.ReadElementContentAsString());
        }
    }
}
```

It throws an exception now, but is safe, isn't it?

_Originally posted by @chuchu in https://github.com/dotnet/roslyn-analyzers/issues/1591#issuecomment-443728167_

----
XmlReader.Create(), without XmlReaderSettings specified, uses the default XmlReaderSettings, which disables DTD processing.

XmlReader.Create(string) still can load XML from external source.  I'm not sure if that was the original motivation for flagging such cases, but I'll leave the existing behavior for now for this overload.

### PR DESCRIPTION
**Pull request by dotpaul**
Fixes #1915 

Not generating diagnostics on:
- XmlReader.Create(Stream)
- XmlReader.Create(TextReader)

because those use the default XmlReaderSettings, which has DtdProcessing set to Prohibit.

### CODE REVIEW

**Comment by dotpaul** [comment]
FYI @timhermann @matt-desai

File: nuget/Microsoft.CodeAnalysis.FxCopAnalyzers/LegacyRulesets/Sdl-7.1.Recommended.Error.ruleset [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0"?>
+<RuleSet Name="Microsoft SDL 7.1 Required and Recommended" Description="This rule set contains all Microsoft SDL 7.1 required and recommended rules, configured to error rather than warn." ToolsVersion="10.0">
```
Comment (dotpaul) [inline nuget/Microsoft.CodeAnalysis.FxCopAnalyzers/LegacyRulesets/Sdl-7.1.Recommended.Error.ruleset]: Errr...I don't remember when I committed the ruleset changes, but they're totally unrelated and should not be part of this pull request.

**Comment by mavasani** [comment]
@dotpaul Can we file a documentation bug at https://docs.microsoft.com/en-us/visualstudio/code-quality/ca3075-insecure-dtd-processing?view=vs-2017 to ensure this is documented, along with examples? Seems like that doc page has lot of extensive examples, so this would be helpful.

File: src/Microsoft.NetFramework.Analyzers/Core/DoNotUseInsecureDtdProcessing.cs [hunk 2]
```diff
@@ -256,15 +256,24 @@ private void AnalyzeMethodOverloads(OperationAnalysisContext context, IMethodSym
...
-                                SecurityDiagnosticHelpers.GetLocalizableResourceString(
-                                    nameof(MicrosoftNetFrameworkAnalyzersResources.XmlReaderCreateWrongOverloadMessage)
-                                )
-                            );
-                        context.ReportDiagnostic(diag);
+                        if (method.Parameters.Length == 1 
+                            && method.Parameters[0].RefKind == RefKind.None
+                            && method.Parameters[0].Type.SpecialType == SpecialType.System_String)
+                        {
+                            // inputUri can load be a URL.  Should further investigate if this is worth flagging.
```
Comment (mavasani) [inline src/Microsoft.NetFramework.Analyzers/Core/DoNotUseInsecureDtdProcessing.cs]: Our docs documentation has a sample that flags this. See second violation here: https://docs.microsoft.com/en-us/visualstudio/code-quality/ca3075-insecure-dtd-processing?view=vs-2017#violations-1

```
using System.Xml;

namespace TestNamespace
{
    class TestClass
    {
        private static void TestMethod()
        {
            var reader = XmlTextReader.Create(""doc.xml""); //warn
        }
    }
}
```

Comment (dotpaul) [reply src/Microsoft.NetFramework.Analyzers/Core/DoNotUseInsecureDtdProcessing.cs]: Right, but I'm not sure what it's supposed to vulnerable to.

Comment (mavasani) [reply src/Microsoft.NetFramework.Analyzers/Core/DoNotUseInsecureDtdProcessing.cs]: Sure, make sure you file a doc bug if you change the semantics here :-)

**Comment by dotpaul** [comment]
https://github.com/MicrosoftDocs/visualstudio-docs-pr/issues/3495 for updating the docs

**Comment by dotpaul** [comment]
Thank you @mavasani !
